### ISSUE TITLE
Parse `$ref`s in JSON

### ISSUE THREAD

**Issue by econchick** [spotify/ramlfications#4]
One can construct JSON pointers using `$ref`s, e.g.

```json
{
  "definitions": {
    "address": {
      "type": "object",
      "properties": {
        "street_address": { "type": "string" },
        "city":           { "type": "string" },
        "state":          { "type": "string" }
      },
      "required": ["street_address", "city", "state"]
    }
  }
}
{ "$ref": "#/definitions/address" }
```

Right now the package doesn't resolve those `$ref`s, and will actually probably need a separate library (maybe [jsonref](https://pypi.python.org/pypi/jsonref)).

**Comment by jenner** [comment]
jsonschema has a [resolver](http://python-jsonschema.readthedocs.org/en/latest/references/) class which can be used to resolve refs relative to schema, i.e. something like { "$ref": "#/definitions/$Thingy/.." } would resolve to an included Thingy schema items

**Comment by econchick** [comment]
@jenner yesssss thanks for the tip!

**Comment by benhamill** [comment]
Would you accept a pull request adding this functionality?

**Comment by econchick** [comment]
@benhamill definitely

**Comment by jenner** [comment]
@benhamill thanks in advance :)

**Comment by econchick** [comment]
Being worked on via #19

**Comment by econchick** [comment]
addressed in #40
